nntrainer/src/flatten_layer.cpp

```diff
@@ -37,22 +37,18 @@ int FlattenLayer::initialize(bool last) {
   return status;
 }
 
-sharedTensor FlattenLayer::forwarding(sharedTensor in) {
-  input = *in;
-
-  hidden = Tensor(input.batch(), output_dim.channel(), output_dim.height(),
-                  output_dim.width());
-  hidden.setZero();
-
-  memcpy(hidden.getData(), input.getData(),
-         input.getDim().getDataLen() * sizeof(float));
-
-  return MAKE_SHARED_TENSOR(hidden);
+sharedConstTensor FlattenLayer::forwarding(sharedConstTensor in) {
+  Tensor ret = in->clone();
+  ret.setDim(output_dim);
```

Comment:
This batch size of this must be input.batch(). not output.batch().

Reply:
Fixed